### PR TITLE
Backend code for creating index and documents

### PR DESCRIPTION
**Pull request by Hodgy**
Code is not yet linked. This is just the backend classes.

### CODE REVIEW

File: includes/class-wp-lucene-search-abstract-request.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<?php
+
+require_once plugin_dir_path( dirname(__FILE__ )) . 'includes/class-wp-lucene-search-request.php';
```
Comment (cooperaj) [inline includes/class-wp-lucene-search-abstract-request.php]: inconsistant spacing

File: includes/class-wp-lucene-search-activator.php [hunk 2]
```diff
@@ -1,36 +0,0 @@
-<?php
```
Comment (cooperaj) [inline includes/class-wp-lucene-search-activator.php]: Should use these activators to set/unset default settings?

Comment (Hodgy) [reply includes/class-wp-lucene-search-activator.php]: There aren't really any default settings to set.

File: includes/class-wp-lucene-search-abstract-request.php (hunk 1 above)
Comment (Hodgy) [reply includes/class-wp-lucene-search-abstract-request.php]: Fixing

**Pull request merged**